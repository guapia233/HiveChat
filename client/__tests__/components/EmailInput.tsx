import { render, screen } from "@testing-library/react";
import EmailInput from "@/components/EmailInput";
import "@testing-library/jest-dom";
import { useForm } from "react-hook-form";
import userEvent from "@testing-library/user-event";

type Inputs = { email: string };

const Form = () => {
  const {
    register,
    formState: { errors },
  } = useForm<Inputs>({ mode: "onChange" });
  return (
    <EmailInput
      register={register}
      name="email"
      errorMessage={errors.email?.message}
    />
  );
};

describe("EmailInput", () => {
  test("Success", async () => {
    const user = userEvent.setup();

    // Render
    const { asFragment } = render(<Form />);

    // Rendered correctly
    expect(screen.getByText("Email")).toBeInTheDocument();

    // Type a value
    await user.type(screen.getByLabelText("email"), "test@test.com");

    // No error
    expect(screen.getByDisplayValue("test@test.com")).toBeInTheDocument();
    expect(screen.queryByText("Please enter a valid email.")).toBeNull();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  test("Invalid email", async () => {
    const user = userEvent.setup();

    // Render
    const { asFragment } = render(<Form />);

    // Type a bad email
    await user.type(screen.getByLabelText("email"), "bad-email");

    // Error message shown
    expect(screen.getByDisplayValue("bad-email")).toBeInTheDocument();
    expect(screen.getByText("Please enter a valid email.")).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });
});
