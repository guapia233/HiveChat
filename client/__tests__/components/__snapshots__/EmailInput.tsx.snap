// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput Invalid email 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r1:"
      id=":r1:-label"
    >
      Email
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk Mui-error css-wgai2y-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-describedby=":r1:-helper-text"
        aria-invalid="true"
        aria-label="email"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r1:"
        maxlength="100"
        name="email"
        required=""
        type="email"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium Mui-focused MuiFormHelperText-filled Mui-required css-1d1r5q-MuiFormHelperText-root"
      id=":r1:-helper-text"
    >
      Please enter a valid email.
    </p>
  </div>
</DocumentFragment>
`;

exports[`EmailInput Success 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for=":r0:"
      id=":r0:-label"
    >
      Email
      <span
        aria-hidden="true"
        class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
      >
         *
      </span>
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-v4u5dn-MuiInputBase-root-MuiInput-root"
    >
      <input
        aria-invalid="false"
        aria-label="email"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        id=":r0:"
        maxlength="100"
        name="email"
        required=""
        type="email"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
